src/router.tsx

```diff
@@ -1,17 +1,23 @@
-import React from 'react';
+import React, { ReactElement } from 'react';
 import { BrowserRouter, Route, Routes } from 'react-router-dom';
 
 import { APP_ROUTES } from './enums';
 import HomeLazy from './pages/home/home-lazy';
 import SigninLazy from './pages/sign-in/sign-in-lazy';
+import { getUserAuthenticated } from './session-storage-utils';
 
-const Router: React.FC = () => (
-  <BrowserRouter>
-    <Routes>
-      <Route path={APP_ROUTES.SIGNIN} element={<SigninLazy />} />
-      <Route path={APP_ROUTES.HOME} element={<HomeLazy />} />
-    </Routes>
-  </BrowserRouter>
-);
+const Router = (): ReactElement => {
+  const isAuthenticated = getUserAuthenticated();
```

Comment:
this is not good approach - use redirect for redirecting to the pages like here https://v5.reactrouter.com/web/example/auth-workflow

Also create 2 different types of route as in example Authporized/NonAuthozied and wrap your pages with them